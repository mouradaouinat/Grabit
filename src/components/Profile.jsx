/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react'
import styles from '../styles/Profile.module.scss'

const Profile = () => {
  return (
    <div className={styles.white}>
      <h1 className={styles.heading}>Profile settings</h1>
      <div className={styles.content}>
        <div className="row">
          <div className="col-lg-6">
            <form>
              <div className="form-group-lg">
                <label htmlFor="name" className={styles.grey}>
                  Full Name
                </label>
                <input
                  type="text"
                  className="form-control mb-4 p-4"
                  id="name"
                  name="name"
                  aria-describedby="emailHelp"
                />
              </div>
              <div className="form-group-lg">
                <label htmlFor="email" className={styles.grey}>
                  Email
                </label>
                <input type="email" className="form-control mb-4 p-4" id="email" />
              </div>
              <div className="form-group-lg">
                <label htmlFor="phone" className={styles.grey}>
                  Phone
                </label>
                <input type="text" className="form-control mb-4 p-4" id="phone" />
              </div>
              <button type="submit" className={styles.button}>
                Update
              </button>
            </form>
          </div>
          <div className="col-lg-6 text-center flex-center">
            <div className={styles.avatar} />
            <div>
              <button type="button" className={styles.btnDark}>
                Upload
              </button>
              <button type="button" className={styles.btnLight}>
                Remove
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Profile
