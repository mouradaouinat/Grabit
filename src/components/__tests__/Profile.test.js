import React from 'react'
import { render } from '@testing-library/react'
import Profile from '../Profile'

describe('Profile', () => {
  it('renders without crashing', () => {
    render(<Profile />)
  })

  it('displays the same content every time it renders', () => {
    const snapshot = render(<Profile />)
    expect(snapshot).toMatchSnapshot()
  })
})
