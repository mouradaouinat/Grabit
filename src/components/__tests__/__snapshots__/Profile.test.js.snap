// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile displays the same content every time it renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="settings__heading"
      >
        Profile Settings
      </h1>
      <div
        class="profile"
      >
        <form>
          <div
            class="row"
          >
            <div
              class="col-lg-6 text-center order-lg-last margin__bottom__mobile--medium"
            >
              <div
                class="profile__avatar profile__avatar--letter"
              />
              <div>
                <input
                  class="profile__upload-button"
                  id="file"
                  name="file"
                  type="file"
                />
                <label
                  for="file"
                >
                  Upload
                </label>
              </div>
            </div>
            <div
              class="col-lg-6"
            >
              <div
                class="form-group-lg"
              >
                <label
                  class="profile__input-label"
                  for="displayName"
                >
                  Name
                </label>
                <input
                  class="form-control p-4 "
                  id="displayName"
                  name="displayName"
                  type="text"
                  value=""
                />
                
              </div>
              <div
                class="form-group-lg mt-4"
              >
                <label
                  class="profile__input-label"
                  for="email"
                >
                  Email
                </label>
                <input
                  class="form-control p-4 "
                  id="email"
                  name="email"
                  type="email"
                  value=""
                />
                
              </div>
              <div
                class="form-group-lg mt-4"
              >
                <label
                  class="profile__input-label"
                  for="phone"
                >
                  Phone
                </label>
                <input
                  class="form-control p-4 "
                  id="phone"
                  name="phone"
                  type="text"
                  value=""
                />
                
              </div>
              <button
                class="button button__block button__primary mt-5"
                type="submit"
              >
                Update
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1
      class="settings__heading"
    >
      Profile Settings
    </h1>
    <div
      class="profile"
    >
      <form>
        <div
          class="row"
        >
          <div
            class="col-lg-6 text-center order-lg-last margin__bottom__mobile--medium"
          >
            <div
              class="profile__avatar profile__avatar--letter"
            />
            <div>
              <input
                class="profile__upload-button"
                id="file"
                name="file"
                type="file"
              />
              <label
                for="file"
              >
                Upload
              </label>
            </div>
          </div>
          <div
            class="col-lg-6"
          >
            <div
              class="form-group-lg"
            >
              <label
                class="profile__input-label"
                for="displayName"
              >
                Name
              </label>
              <input
                class="form-control p-4 "
                id="displayName"
                name="displayName"
                type="text"
                value=""
              />
              
            </div>
            <div
              class="form-group-lg mt-4"
            >
              <label
                class="profile__input-label"
                for="email"
              >
                Email
              </label>
              <input
                class="form-control p-4 "
                id="email"
                name="email"
                type="email"
                value=""
              />
              
            </div>
            <div
              class="form-group-lg mt-4"
            >
              <label
                class="profile__input-label"
                for="phone"
              >
                Phone
              </label>
              <input
                class="form-control p-4 "
                id="phone"
                name="phone"
                type="text"
                value=""
              />
              
            </div>
            <button
              class="button button__block button__primary mt-5"
              type="submit"
            >
              Update
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
